src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -2841,6 +2841,45 @@ exprt c_typecheck_baset::do_special_functions(
     overflow.type() = bool_typet{};
     return overflow;
   }
+  else if(identifier == CPROVER_PREFIX "enum_is_in_range")
+  {
+    // Check correct number of arguments
+    if(expr.arguments().size() != 1)
+    {
+      error().source_location = expr.source_location();
+      error() << identifier << " expects one argument, "
+              << expr.arguments().size() << " were provided" << eom;
+      throw 0;
+    }
+    // Replace expr with giant boolean that checks all the values
+    // of the enumeration as disjunction
+    // This code is similar to goto_checkt::enum_range_check
+    auto arg1 = expr.arguments()[0];
+    typecheck_expr(arg1);
+    if(
+      auto c_enum_tag_type =
+        type_try_dynamic_cast<c_enum_tag_typet>(arg1.type()))
+    {
+      const c_enum_typet &c_enum_type = follow_tag(*c_enum_tag_type);
+      const c_enum_typet::memberst enum_values = c_enum_type.members();
+
+      std::vector<exprt> disjuncts;
+      for(const auto &enum_value : enum_values)
+      {
+        const constant_exprt val{enum_value.get_value(), *c_enum_tag_type};
+        disjuncts.push_back(equal_exprt(arg1, val));
+      }
+
+      return disjunction(disjuncts);
+    }
+    else
+    {
+      // Can't enum range check a non-enum
+      error().source_location = expr.source_location();
+      error() << identifier << " expects enum argument type" << eom;
```

Comment:
Again, would be better if the error message also stated what we got _instead_ of what it expected.

Reply:
Fixed.